sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1252,7 +1252,9 @@ class Analyzer(
           case ae: AnalysisException => filter
         }
 
-      case sort @ Sort(sortOrder, global, aggregate: Aggregate) if aggregate.resolved =>
+      case sort @ Sort(sortOrder, global, aggregate: Aggregate)
+        if aggregate.resolved && sortOrder.forall(x => IntegerIndex.unapply(x.child).isEmpty) =>
```

Comment:
Have you tried to move this extra check into `resolved` of `Aggregate` operator? Does it work?